ledger/src/blockstore_processor.rs

```diff
@@ -194,13 +195,14 @@ fn execute_batch(
         let remaining_block_cost_cap = cost_capacity_meter
             .write()
             .unwrap()
-            .accumulate(execution_cost_units);
+            .accumulate(execution_cost_units + tx_cost);
```

Comment:
what do you think about checking account limit? This currently checks against block limit (48M CUs per block), but not checking if single writable account limit of 12M CUs. In the worst scenario, if all transactions write to same account, the block would take 4 times longer to process.

Reply:
That would be a chunk of new code? Or is the leader already doing this? I'd assume it would require a hashmap from key to a shared/atomic sum for that account? Could add a fair amount of expense to this check, and it's at least not in the "use case" for why a non-malicious leader would want to cheat (which is "I make more money if I distribute larger blocks").

Reply:
Looks like https://github.com/solana-labs/solana/issues/24657 may essentially handle that? (when implemented)

Reply:
yeah, leader uses `cost_tracker` to check both account and block limits.

Reply:
> Looks like #24657 may essentially handle that? (when implemented)

oh, that's a different function - that is to drop packets paying less than minimum required fee for contented account before it even get into banking_stage. 

Here we want to protect validators from replaying blocker longer than 400ms ( if a block has more than 12m CU on same account, then it is likely to take more than 400ms to replay)

Reply:
Tracking this issue in https://github.com/solana-labs/solana/issues/25166